### PR TITLE
[SUPPORTESC-109] Add details about when dynamic fields are loaded

### PR DESCRIPTION
**Pull request by armchairlinguist**
This docs addition provides more guidance to developers about use contexts when they are working with code for loading dynamic fields.

Related to https://github.com/zapier/zapier-platform/pull/303.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,7 +1,2 @@
-_site/
-.sass-cache/
-.jekyll-cache/
-.jekyll-metadata
-.DS_Store
-docs/.DS_Store
-.vscode
+.gitignore
+.ruby-version
```
Comment (codebycaleb) [inline .gitignore:2]: Did you mean to remove the other lines in this file?

Comment (armchairlinguist) [reply .gitignore:2]: Oh heck. I didn't mean to commit this file at all! Thanks for catching that.

Comment (armchairlinguist) [reply .gitignore]: Should be sorted out now!

**Review by codebycaleb — APPROVED** [review]
Thanks for thinking of platform.zapier.com too! 📖 ❤️

**Pull request merged**